### PR TITLE
Simplify pytests

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Makes pytest work by just running `pytest` in the root directory

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

**Review by anthonylasso — APPROVED** [review]
confirmed. can rust tests from the root. 
Screenshot 2023-12-14 at 4 48 45 PM

File: client/src/featureform/providers/filestore.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def flush(self):
 
 
 class LocalFileStore(OnlineStore):
-    def __init__(self, config):
+    def __init__(self, config=""):
```
Comment (aolfat) [inline client/src/featureform/providers/filestore.py:82]: why'd you have to do this? should it default to None?

File: client/src/featureform/sqlite_metadata.py [hunk 2]
```diff
@@ -496,9 +496,12 @@ def get_model(self, name, should_fetch_tags_properties):
...
     def get_provider(self, name, should_fetch_tags_properties):
-        return self.query_resource(
-            "providers", "name", name, should_fetch_tags_properties
-        )[0]
+        try:
+            return self.query_resource(
+                "providers", "name", name, should_fetch_tags_properties
+            )[0]
+        except:
+            raise ValueError(f"Provider {name} not found")
```
Comment (aolfat) [inline client/src/featureform/sqlite_metadata.py:504]: nit: is this the only reason this throws an error?

File: client/tests/test_online.py [hunk 3]
```diff
@@ -56,6 +56,7 @@ def get_random_feature_variant():
 ]
 
 
+@pytest.mark.skip(reason="out of scope for now")
```
Comment (aolfat) [inline client/tests/test_online.py:59]: are these important tests?

File: provider/scripts/k8s/tests/test_offline_store_pandas_runner.py [hunk 4]
```diff
@@ -179,6 +179,7 @@ def test_get_blob_credentials(variables, type, request):
     assert credentials == expected_output
 
 
+@pytest.mark.skip("Requires actual connection strings")
```
Comment (aolfat) [inline provider/scripts/k8s/tests/test_offline_store_pandas_runner.py:182]: is it possible to still allow these to run but with a certain flag?

File: pytest.ini [hunk 5]
```diff
@@ -3,3 +3,4 @@ markers =
     local: tests to execute in local mode
     hosted: tests to execute in hosted mode
     docker: tests to execute in docker
+addopts = -m "not docker and not hosted"
```
Comment (aolfat) [inline pytest.ini:6]: what is diss? addopts vs adopts?

File: Makefile [hunk 6]
```diff
@@ -210,39 +210,7 @@ stop_postgres:
...
-	python -m pytest -m 'local' client/tests/test_class_api.py
-	python -m pytest -m 'local' client/tests/test_ondemand_features.py
-	python -m pytest -m 'local' client/tests/test_resource_registration.py
-	python -m pytest -m 'local' client/tests/test_source_dataframe.py
-	python -m pytest -m 'local' client/tests/test_training_set_dataframe.py
-	python -m pytest -m 'local' client/tests/get_provider_test.py
-	python -m pytest client/tests/test_deploy.py
-	-rm -r .featureform
-	-rm -f transactions.csv
+	python -m pytest
```
Comment (aolfat) [inline Makefile:213]: what happens with the local and server stuff?

File: client/src/featureform/sqlite_metadata.py (hunk 2 above)
Comment (sdreyer) [reply client/src/featureform/sqlite_metadata.py:504]: The only other time should be if it cant connect to the database. Can add another case

File: Makefile (hunk 6 above)
Comment (sdreyer) [reply Makefile:213]: So by default now it runs the local with that addopts flag in the conf file. You can run hosted and docker manually by running it with the those flags but we dont do that here

File: provider/scripts/k8s/tests/test_offline_store_pandas_runner.py (hunk 4 above)
Comment (sdreyer) [reply provider/scripts/k8s/tests/test_offline_store_pandas_runner.py]: Changed

**Pull request merged**